uswds/uswds
# [Test] Prose placeholder fix

## Summary
Fixed compilation issues related to `usa-prose-` placeholders. 

## Breaking change
:warning: This is potentially a breaking change. This PR removes `usa-prose-` mixins and placeholders. Users should instead use `typeset-` mixins in their place. 

## Related issue


## Preview link
Preview link: [Prose component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-prose-fix/?path=/story/components-prose--default)

## Problem statement
From https://github.com/uswds/uswds/issues/4962: 
> I’m trying to configure USWDS inside an @layer rule using the load-css function from sass:meta. When I try to @use "uswds-core" in my own Sass modules so that I can access the tokens for my styles, I get a target selector not found error for %usa-prose-p.


## Solution
Sass seems to get confused when `* + &` is brought in from another file via a mixin (more details in https://github.com/uswds/uswds/pull/4576). Additionally, Sass can throw errors in limited circumstances when we rely on `@extend` and placeholders.

The key conclusion is that `* + &` does not currently seem to be a viable addition to mixins or placeholders. By moving all `* + &` out of mixins and placeholders and directly into the style definitions, we can remove the risk of Sass confusion.

Note: The downside of this solution is that it requires some code repetition, but that might be slightly mitigated by creating variables/settings for the heading and paragraph top margins. 

## Tasks completed
1. Moved * + & references out of mixins/placeholders and directly into the style definitions.
1. Removed redundant and unused `usa-prose-` mixins and replaced them with `typeset-` mixins. 

## Testing and review
1. Confirm that Sass compiles without error
1. Confirm that 

This will need more testing, but I experimented with moving `* + &` references out of mixins/placeholders and directly into the style definitions. 

This was done because:
1. Sass seems to get confused when `* + &` is brought in from another file via a mixin (more details in https://github.com/uswds/uswds/pull/4576).
3. Sass can throw errors in limited circumstances when we rely on `@extend` and placeholders.

This PR represents only one possible path for moving forward. The downside is that it requires some code repetition, but that might be slightly mitigated by creating variables/settings for the heading and paragraph top margins. There are likely other CSS/Sass solutions that can generate the same result. 

Additionally, this PR in its current state removes unused mixins and placeholders, but we might want to retain those in case users might be calling these items in their projects.

The key conclusion is that `* + &` does not currently seem to be a viable addition to mixins or placeholders.
